### PR TITLE
Add nix tabry lang

### PR DESCRIPTION
**Pull request by pfgray**
This commit adds a nix derivation that builds the tabry compiler (a treesitter application). It also adds a flake to make invoking with nix easy.

These PRs are all related:

_ [Add nix tabry hm module](https://github.com/evanbattaglia/tabry/pull/3)
X [Add nix tabry lang](https://github.com/evanbattaglia/tabry/pull/2)
_ [Add nix derivation for tabry](https://github.com/evanbattaglia/tabry/pull/4)

### CODE REVIEW

File: treesitter/default.nix [hunk 1]
```diff
@@ -0,0 +1,79 @@
+flake-utils: {
+  yarn2nix,
+  yarn2nix-moretea,
```
Comment (evanbattaglia) [inline treesitter/default.nix]: I don't see the following packages used, are they leftover from testing or do mkYarnPackage use them somehow?

yarn2nix
yarn2nix-moretea
nodePackages

The following also aren't used. Tabry is ruby but only compiling shouldn't need it, unless you're including it for another reason?

ruby
bundlerEnv

I also don't see "fetchzip" used?

File: treesitter/default.nix [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    formatJsonFilename = fileName: 
+      (commandNameFromTabryFilename fileName) + ".json";
+
+    tabryCompilerJsPath = "/libexec/tree-sitter-tabry/deps/tree-sitter-tabry/tabry-compile.js";
+
+    # This is a function that takes a .tabry file
+    # and returns a derivation that compiles that
+    # .tabry file into the tabry .json file
+    compileTabryFile = inFile: stdenv.mkDerivation {
```
Comment (evanbattaglia) [inline treesitter/default.nix:27]: This isn't used anywhere else in the commit; what's the use case for it?

**Review by evanbattaglia — COMMENTED** [review]
I have a couple questions but it looks great, thanks!

File: treesitter/default.nix (hunk 1 above)
Comment (pfgray) [reply treesitter/default.nix]: No, you're right, this is all stuff leftover from when I was debugging... I'll clean it up!

File: treesitter/default.nix (hunk 2 above)
Comment (pfgray) [reply treesitter/default.nix:27]: It's used in https://github.com/evanbattaglia/tabry/pull/3 to compile `.tabry` files from people's home-manager config,

for example:
```
config.programs.tabry = {
  enable = true;
  tabryFiles = [
    ./rapture.tabry
    ./aws.tabry
  ]
}
```

this function is useful for compiling `rapture.tabry` and `aws.tabry`

**Pull request merged**